frontend/src/metabase/admin/datamodel/utils/humanizeCoercionStrategy.ts

```diff
@@ -0,0 +1,43 @@
+const LEFT_TERM_CONVERSIONS: Record<string, string> = {
+  ISO8601: "ISO 8601",
+  UNIXSeconds: "UNIX seconds",
+  UNIXMilliSeconds: "UNIX milliseconds",
+  UNIXMicroSeconds: "UNIX microseconds",
+  UNIXNanoSeconds: "UNIX nanoseconds",
+  YYYYMMDDHHMMSSString: "YYYYMMDDHHMMSS string",
+  YYYYMMDDHHMMSSBytes: "YYYYMMDDHHMMSS bytes",
+};
+
+const RIGHT_TERM_CONVERSIONS: Record<string, string> = {
+  DateTime: "Datetime",
+};
+
+/**
+ * Converts -> to → and humanizes strings
+ * @param {string} fullString - The coercion strategy as it comes from the back-end
+ * @returns {string}
+ */
+export function humanizeCoercionStrategy(fullString: string) {
+  const shortString = fullString.replace("Coercion/", "");
+
+  const [leftTerm, rightTerm] = shortString.split("->");
+
+  return rightTerm === undefined
+    ? shortString
+    : treatTermsAndJoin(leftTerm, rightTerm);
+}
+
+function treatTermsAndJoin(left: string, right: string) {
+  const treatedLeftTerm = treatLeftTerm(left);
+  const treatedRightTerm = treatRightTerm(right);
+
+  return [treatedLeftTerm, treatedRightTerm].join(" → ");
```

Comment:
I'll leave this up to you but strictly speaking it would be smart to localize the arrow too (with a context string explaining what it means here). Otherwise it will point backwards in RTL languages which will be super confusing